screens/AppNavigator/screens/BalancesScreen/SendScreen/SendScreen.tsx

```diff
@@ -0,0 +1,206 @@
+import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
+import { MaterialIcons } from '@expo/vector-icons'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useState } from 'react'
+import { ScrollView, TouchableOpacity, View, StyleSheet, Modal } from 'react-native'
+import { useForm, Controller, Control } from 'react-hook-form'
+import { BarCodeScanner } from 'expo-barcode-scanner'
+import NumberFormat from 'react-number-format'
+import tailwind from 'tailwind-rn'
+import { PrimaryButton, Text, TextInput } from '../../../../../components'
+import { getTokenIcon } from '../../../../../components/icons/tokens/_index'
+import { PrimaryColor, PrimaryColorStyle } from '../../../../../constants/Theme'
+import { translate } from '../../../../../translations'
+import { BalanceParamList } from '../BalancesNavigator'
+
+interface QRScannerProps {
+  onScan: (code: string) => void
+  isQRScanning: boolean
+  onCloseModal: () => void
+}
+
+interface AmountForm {
+  control: Control
+  token: AddressToken
+  onMaxPress: (amount: string) => void
+}
+
+interface AddressForm {
+  control: Control
+  openQRScanner: () => void
+}
+
+type Props = StackScreenProps<BalanceParamList, 'SendScreen'>
+
+export function SendScreen ({ route }: Props): JSX.Element {
+  const [token] = useState(route.params.token)
+  const [isQRScanning, setIsQRScanning] = useState(false)
+  const { control, setValue, formState: { isValid }, getValues, trigger } = useForm({ mode: 'onChange' })
+
+  const onSubmit = (): void => {
+    console.log(isValid)
+    console.log(getValues())
+  }
+
+  return (
+    <ScrollView style={tailwind('bg-gray-100')}>
+      <QRScanner
+        onScan={async (value: string) => {
+          setValue('address', value)
+          await trigger('address')
+          setIsQRScanning(false)
+        }} onCloseModal={() => setIsQRScanning(false)} isQRScanning={isQRScanning}
+      />
+      <AddressRow control={control} openQRScanner={() => setIsQRScanning(true)} />
+      <AmountRow
+        token={token} control={control} onMaxPress={async (amount) => {
+          setValue('amount', amount)
+          await trigger('amount')
+        }}
+      />
+      <PrimaryButton disabled={!isValid} title='Send' onPress={onSubmit}>
```

Comment:
Not familiar with this `useForm` API, but I cant identify what's the logic that decide whether it `isValid` here

Reply:
@ivan-zynesis `useForm` API is from here - https://www.react-hook-form.com/. Basically, it's a hook to do form validations, like required, max amount and number pattern.